### PR TITLE
Initialize FastHotStuffConsensus when node is a validator

### PR DESCRIPTION
**Pull request by tholonious**

### CODE REVIEW

**Comment by tholonious** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/deso-protocol/core/906?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `feature/proof-of-stake`
  * **PR #875** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/875?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #889** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/889?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #906** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/906?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/906?utm_source=stack-comment).

File: cmd/run.go [hunk 1]
```diff
@@ -58,6 +58,10 @@ func SetupRunFlags(cmd *cobra.Command) {
 	cmd.PersistentFlags().String("postgres-uri", "", "BETA: Use Postgres as the backing store for chain data."+
 		"When enabled, most data is stored in postgres although badger is still currently used for some state. Run your "+
 		"Postgres instance on the same machine as your node for optimal performance.")
+	cmd.PersistentFlags().String("pos-validator-seed", "", "The private key of the Proof of Stake validator. "+
+		"The private key should be passed as hex, optionally prefixed with a '0x', and map to a valid BLS12_381 private key. "+
```
Comment (lazynina) [inline cmd/run.go:62]: Do we want a private key hex or a seed phrase? We should just be clear about how we want this passed in. should we call this `pos-validator-seed-hex` if we want a hex?

File: cmd/run.go [hunk 2]
```diff
@@ -58,6 +58,10 @@ func SetupRunFlags(cmd *cobra.Command) {
...
+		"The private key must be 32 bytes, or 64 characters, in length (excluding the '0x' prefix). Setting this flag automatically "+
+		"makes the node run Proof of Stake Validator.")
```
Comment (lazynina) [inline cmd/run.go:64]:
```suggestion
		"makes the node run as a Proof of Stake Validator.")
```

File: lib/pos_consensus.go [hunk 3]
```diff
@@ -31,7 +31,7 @@ func NewFastHotStuffConsensus(params *DeSoParams, blockchain *Blockchain, mempoo
 // blockchain state. This should only be called once the blockchain has synced, the node is
 // ready to join the validator network, and the node is able to validate blocks in the steady state.
 func (cc *FastHotStuffConsensus) Start() error {
-	// Hold the write consensus's lock for thread-safety.
+	// Hold the consensus's write lock for thread-safety.
```
Comment (lazynina) [inline lib/pos_consensus.go:34]:
```suggestion
	// Hold the consensus' write lock for thread-safety.
```
grammar nit 😂

**Review by lazynina — CHANGES_REQUESTED** [review]

File: cmd/run.go (hunk 1 above)
Comment (tholonious) [reply cmd/run.go:62]: Sure. Adding the "hex" suffix makes it clearer

Comment (lazynina) [reply cmd/run.go:62]: Agreed!!

File: cmd/run.go [hunk 4]
```diff
@@ -58,6 +58,10 @@ func SetupRunFlags(cmd *cobra.Command) {
 	cmd.PersistentFlags().String("postgres-uri", "", "BETA: Use Postgres as the backing store for chain data."+
 		"When enabled, most data is stored in postgres although badger is still currently used for some state. Run your "+
 		"Postgres instance on the same machine as your node for optimal performance.")
+	cmd.PersistentFlags().String("pos-validator-seed", "", "The private key of the Proof of Stake validator. "+
```
Comment (lazynina) [inline cmd/run.go:61]:
```suggestion
	cmd.PersistentFlags().String("pos-validator-seed-hex", "", "The private key of the Proof of Stake validator. "+
```

**Review by lazynina — APPROVED** [review]

**Review by lazynina — APPROVED** [review]

**Pull request merged**